OMEdit/OMEditGUI/Editors/BaseEditor.cpp

```diff
@@ -1390,6 +1425,31 @@ void PlainTextEdit::resizeEvent(QResizeEvent *pEvent)
   mpLineNumberArea->setGeometry(QRect(cr.left(), cr.top(), lineNumberAreaWidth(), cr.height()));
 }
 
+QCompleter *PlainTextEdit::completer()
+{
+  return mpcompleter;
+}
+
+/* insert the keywords from the completer popup */
+void PlainTextEdit::insertCompletion(const QString& completion)
+{
+  if (mpcompleter->widget() != this)
+      return;
+  QTextCursor cursor = textCursor();
+  int extra = completion.length() - mpcompleter->completionPrefix().length();
+  //cursor.movePosition(QTextCursor::Left);
+  cursor.movePosition(QTextCursor::EndOfWord);
+  cursor.insertText(completion.right(extra));
+  setTextCursor(cursor);
+}
+
+QString PlainTextEdit::textUnderCursor() const
+{
+  QTextCursor cursor = textCursor();
+  cursor.select(QTextCursor::WordUnderCursor);
+  return cursor.selectedText();
+}
```

Comment:
This function seems fishy.....Looks like the text remained selected after this function call.
Shouldn't we clear the selection after reading the text?